mfr/extensions/audio/render.py

```diff
@@ -13,7 +14,8 @@ class AudioRenderer(extension.BaseRenderer):
         ]).get_template('viewer.mako')
 
     def render(self):
-        return self.TEMPLATE.render(base=self.assets_url, url=self.url)
+        safe_url = escape_url_for_template(self.url)
+        return self.TEMPLATE.render(base=self.assets_url, url=safe_url)
```

Comment:
Same fix for all eligible extensions, just different URLs. I can change the style to
```python
self.TEMPLATE.render(base=self.assets_url, url=escape_url_for_template(...))
``` 
for this and other extensions if you prefer.